src/main/java/org/springframework/data/domain/ExampleSpecAccessor.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.domain;
+
+import java.util.Collection;
+
+/**
+ * Accessor for the {@link ExampleSpec} to use in modules that support query by example (QBE) querying.
+ *
+ * @author Mark Paluch
+ * @since 1.12
+ */
+public class ExampleSpecAccessor {
```

Comment:
This class is used by the individual modules to create queries and to access details stored in Example/ExampleSpec. How would have solved it? Put it to query.spi and add public getters to ExampleSpec?